packages/liveblocks-core/src/__tests__/room.test.ts

```diff
@@ -1276,6 +1276,45 @@ describe("room", () => {
         },
       ]);
     });
+
+    test("hasPendingStorageModifications", async () => {
+      const { storage, assert, machine, refStorage, reconnect, ws } =
+        await prepareStorageTest<{ x: number }>(
+          [createSerializedObject("0:0", { x: 0 })],
+          1
+        );
+
+      assert({ x: 0 });
+
+      expect(machine.hasPendingStorageModifications()).toBe(false);
+
+      ws.closeFromBackend(
+        new CloseEvent("close", {
+          code: WebsocketCloseCodes.CLOSE_ABNORMAL,
+          wasClean: false,
+        })
+      );
+
+      storage.root.set("x", 1);
+
+      expect(machine.hasPendingStorageModifications()).toBe(true);
+
+      const storageJson = lsonToJson(storage.root);
+      expect(storageJson).toEqual({ x: 1 });
+      const refStorageJson = lsonToJson(refStorage.root);
+      expect(refStorageJson).toEqual({ x: 0 });
+
+      const newInitStorage: IdTuple<SerializedCrdt>[] = [
+        createSerializedObject("0:0", { x: 0 }),
+      ];
+
+      reconnect(2, newInitStorage);
+
+      assert({
+        x: 1,
+      });
+      expect(machine.hasPendingStorageModifications()).toBe(false);
```

Comment:
A little surprised that `reconnect()` is synchronous so that this even works. I would expect we'd have to `await` it to be able to send the offlineOps?

Reply:
In this context, `reconnect` is a synchronous test utility. We send the initial storage after reconnecting, and the client sends the diff synchronously.